nncf/common/quantization/quantizer_propagation/grouping.py

```diff
@@ -0,0 +1,102 @@
+"""
+ Copyright (c) 2021 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from copy import copy
+from typing import Dict
+from typing import Optional
+from typing import Set
+
+from nncf.common.quantization.quantizer_propagation.structs import PropagatingQuantizer
+
+
+class UnifiedScalePropagatingQuantizerGroupManager:
+    """
+    Keeps track of the groups of quantizers that have to have their scales unified in the final
+    quantized model.
+    """
+    def __init__(self):
+        self._next_gid = 0
+        self._group_vs_prop_quants_dict = {}  # type: Dict[int, Set[PropagatingQuantizer]]
+
+    def _get_next_gid(self) -> int:
+        retval = self._next_gid
+        self._next_gid += 1
+        return retval
+
+    def register_group(self, prop_quants: Set[PropagatingQuantizer]) -> int:
```

Comment:
Could you please add docstrings for all "public" class methods?

Reply:
Done